esp-hal/src/uart.rs

```diff
@@ -2851,6 +2918,97 @@ impl Info {
+                cfg_if::cfg_if! {
+                    if #[cfg(any(esp32c6, esp32h2))] {
+                        self.regs().swfc_conf0().modify(|_, w| w.xonoff_del().set_bit());
+                        self.regs().swfc_conf0().modify(|_, w| w.sw_flow_con_en().set_bit());
+                        self.regs().swfc_conf1().modify(|_, w| unsafe { w.xon_threshold().bits(xon_threshold) });
+                        self.regs().swfc_conf1().modify(|_, w| unsafe { w.xoff_threshold().bits(xoff_threshold) });
+                        self.regs().swfc_conf0().modify(|_, w| unsafe { w.xon_char().bits(xon_char) });
+                        self.regs().swfc_conf0().modify(|_, w| unsafe { w.xoff_char().bits(xoff_char) });
+                    } else if #[cfg(esp32)]{
+                        self.regs().flow_conf().modify(|_, w| w.xonoff_del().set_bit());
+                        self.regs().flow_conf().modify(|_, w| w.sw_flow_con_en().set_bit());
+                        self.regs().swfc_conf().modify(|_, w| unsafe { w.xon_threshold().bits(xon_threshold) });
+                        self.regs().swfc_conf().modify(|_, w| unsafe { w.xoff_threshold().bits(xoff_threshold) });
+                        self.regs().swfc_conf().modify(|_, w| unsafe { w.xon_char().bits(xon_char) });
+                        self.regs().swfc_conf().modify(|_, w| unsafe { w.xoff_char().bits(xoff_char) });
+                    } else {
+                        self.regs().flow_conf().modify(|_, w| w.xonoff_del().set_bit());
+                        self.regs().flow_conf().modify(|_, w| w.sw_flow_con_en().set_bit());
+                        self.regs().swfc_conf1().modify(|_, w| unsafe { w.xon_threshold().bits(xon_threshold as u16) });
+                        self.regs().swfc_conf0().modify(|_, w| unsafe { w.xoff_threshold().bits(xoff_threshold as u16) });
+                        self.regs().swfc_conf1().modify(|_, w| unsafe { w.xon_char().bits(xon_char) });
+                        self.regs().swfc_conf0().modify(|_, w| unsafe { w.xoff_char().bits(xoff_char) });
+                    }
+                }
+            }
+            SwFlowControl::Disabled => {
+                cfg_if::cfg_if! {
+                    if #[cfg(any(esp32c6, esp32h2))] {
+                        let reg = self.regs().swfc_conf0();
+                    } else {
+                        let reg = self.regs().flow_conf();
+                    }
+                }
+
+                reg.modify(|_, w| w.sw_flow_con_en().clear_bit());
+                reg.modify(|_, w| w.xonoff_del().clear_bit());
+            }
+        }
+
+        self.set_cts(matches!(hw_flow_ctrl.cts, CtsConfig::Enabled));
+
+        match hw_flow_ctrl.rts {
+            RtsConfig::Enabled(threshold) => self.set_rts(true, Some(threshold)),
+            RtsConfig::Disabled => self.set_rts(false, None),
+        }
+
+        #[cfg(any(esp32c6, esp32h2))]
+        sync_regs(self.regs());
+    }
+
+    fn set_cts(&self, enable: bool) {
```

Comment:
This sound to me like it's setting the pin high, not enabling tx flow control. I'd suggest finding another name for these